microprofile-fault-tolerance/src/main/java/org/wildfly/quickstarts/microprofile/faulttolerance/CoffeeResource.java

```diff
@@ -0,0 +1,167 @@
+import org.jboss.logging.Logger;
+import org.jboss.resteasy.annotations.jaxrs.PathParam;
+
+/**
+ * A JAX-RS resource that provides information about kinds of coffees we have on store and numbers of packages
+ * available.
+ *
+ * Demonstrates {@link Retry}, {@link CircuitBreaker}, {@link Timeout} and {@link Fallback} policies.
+ */
+@Path("/coffee")
+@Produces(MediaType.APPLICATION_JSON)
+public class CoffeeResource {
+
+    private static final Logger LOGGER = Logger.getLogger(CoffeeResource.class);
+
+    @Inject
+    CoffeeRepositoryService coffeeRepository;
+
+    private AtomicLong counter = new AtomicLong(0);
+
+    private Float failRatio = 0.5f;
+
+    /**
+     * Provides list of all our coffees.
+     *
+     * This method fails about 50% of time. However, in case of a failure, the method is automatically re-invoked again
+     * (up to 4 times), thanks to the {@link Retry} annotation. That means that a user is rarely exposed to a
+     * failure, since the probability of a failure occurring 4 times in row is fairly low.
+     */
+    @GET
+    @Retry(maxRetries = 4, retryOn = RuntimeException.class)
+    public List<Coffee> coffees() {
+        final Long invocationNumber = counter.getAndIncrement();
+
+        maybeFail(String.format("CoffeeResource#coffees() invocation #%d failed", invocationNumber));
+
+        LOGGER.infof("CoffeeResource#coffees() invocation #%d returning successfully", invocationNumber);
+        return coffeeRepository.getAllCoffees();
+    }
+
+    /**
+     * Provides information about a coffee with given id.
+     *
+     * Same as the {@link #coffees()} method, this method fails about 50% of time. However, because this method doesn't
+     * have any {@link Retry} policy, a user is exposed to the failures every time they happen.
+     */
+    @Path("/{id}")
+    @GET
+    public Response coffeeDetail(@PathParam int id) {
+        final Long invocationNumber = counter.getAndIncrement();
+
+        maybeFail(String.format("CoffeeResource#coffees() invocation #%d failed", invocationNumber));
+
+        LOGGER.infof("CoffeeResource#coffees() invocation #%d returning successfully", invocationNumber);
+        Coffee coffee = coffeeRepository.getCoffeeById(id);
+
+        // if coffee with given id not found, return 404
+        if (coffee == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+
+        return Response.ok(coffee).build();
+    }
+
+    /**
+     * Returns how many packages of given coffee we have on store.
+     *
+     * In this case, the failures are introduced inside the CDI bean this method calls. The bean will repeat two
+     * successful and two failed invocations. It also defines a {@link CircuitBreaker} policy, which will open a circuit
+     * breaker after two failed invocations, and then stay open for 5 seconds.
+     */
+    @Path("/{id}/availability")
+    @GET
+    public Response availability(@PathParam int id) {
```

Comment:
`@PathParam` -> `@PathParam("id")`

This fixes Circuit Breaker issue that I mentioned. 

Now, the id is always 0 and there is no coffee with id == 0. They have ids 1, 2 and 3. In other usages of PathParam, id is 0 as well but it doesn't break functionality of the quickstart entirely. For example recommendations(@PathParam int id) method calls coffeeRepository.getRecommendations(id) which only checks coffees by non-equality of the id so it always picks first two coffees to show. Correctly, it should show coffees 1 and 3 if called with id = 2.

That being said, this code is technically correct, but it requires some additional configuration, as mentioned here:

https://docs.jboss.org/resteasy/docs/4.5.2.Final/userguide/html_single/index.html#_NewParam

Reply:
Good point, talked to RESTeasy team, we don't need to use the fancy autogenerated stuff. Done now.